### DISCUSSION TITLE
Cannot connect to port 7000 when switch to Rancher Desktop

### DISCUSSION THREAD

**Discussion by abialas** [Grokzen/docker-redis-cluster#137] [Q&A] [answered]
Hi all,

I had to switch on my Mac machine from Docker Desktop to Rancher Desktop. 
When I was running my tests on Docker Desktop all worked fine. When I switch to Rancher Desktop I cannot connect to port 7000 (all others work fine). I am using cluster with test-containers. I have exposed all ports (7000-7005) and I added IP: 0.0.0.0.

Any idea what must be changed or how to verify what is missing?

**Comment by devaljain1998** [comment]
@Grokzen can you please address this?

**Comment by Grokzen** [comment, marked as answer]
@devaljain1998 @abialas Addressing it how? This is the most common issue with redis-cluster on the same machine and i have written about it in 10 other issues before.

The short answer is that redis-cluster tracks and manages the IP addresses it needs in order to communicate with all other nodes within the cluster. If the nodes would be on different docker containers then the IP would be the public IP for each container. But with this docker image where all nodes is on the same machine, then redis itself tracks the address 127.0.0.1 or sometimes 0.0.0.0 as the IP it needs to communicate with the other cluster nodes.

The issue is not redis tho, but all clients connecting into it. The redis clients is built in sucha way that it uses the same IP mapping that the cluster itself uses in order for all clients to communicate with all nodes correct. But if the redis-cluster says, oh all nodes is on 127.0.0.1 then all clients will think, oh all nodes is on 127.0.0.1:7000-7005, but i can't connect there as docker is in the way. This is why a host network is better in this setup as it would expose the ports directly to your host, but then you can't connect from another machine.

I have no idea how Rancher Desktop works, or what other different magic it do to break this. The most common way to diagnose this is to go into the running docker container, connect with redis-cli into the cluster nodes and run `CLUSTER INFO` & `CLUSTER SLOTS` and try to figure out what IP address that redis things all nodes is at and then properly expose your internal container network to the outside for clients to start to work.

But i simply just use a host-network and that have always worked for me, i suggest you do the same and there is not really anything i am going to change in the docker image itself to change any of this tho
